### PR TITLE
Delete authorization informers

### PR DESCRIPTION
**Pull request by enj**
These resources are just proxies to the RBAC resources now.  Nothing calls these informers directly anymore.

Signed-off-by: Monis Khan <mkhan@redhat.com>

/assign @deads2k @simo5

### CODE REVIEW

File: pkg/authorization/generated/informers/externalversions/authorization/v1/interface.go [hunk 1]
```diff
@@ -1,83 +0,0 @@
...
-	// Policies returns a PolicyInformer.
-	Policies() PolicyInformer
-	// PolicyBindings returns a PolicyBindingInformer.
-	PolicyBindings() PolicyBindingInformer
-	// Roles returns a RoleInformer.
-	Roles() RoleInformer
-	// RoleBindings returns a RoleBindingInformer.
-	RoleBindings() RoleBindingInformer
-	// RoleBindingRestrictions returns a RoleBindingRestrictionInformer.
-	RoleBindingRestrictions() RoleBindingRestrictionInformer
```
Comment (enj) [inline pkg/authorization/generated/informers/externalversions/authorization/v1/interface.go]: This is the only resource that could ever need an informer in the future.

**Comment by enj** [comment]
/hold

3.8

**Comment by enj** [comment]
Now that I think about it, we may need this Informer for the OAuth client.

File: pkg/authorization/generated/informers/externalversions/authorization/v1/interface.go (hunk 1 above)
Comment (simo5) [reply pkg/authorization/generated/informers/externalversions/authorization/v1/interface.go]: I was wondering why you were deleting that one too indeed.

**Comment by liggitt** [comment]
In the rebase, seeing perpetual attempts to watch oauth resources from the GC controller, which always fail. Is that happening in 3.7?

**Comment by enj** [comment]
@liggitt I don't think so, but cannot say for sure.  The token timeout PR uses the OAuth shared informer without issue, so watching those should work fine.  This informer is unrelated to the OAuth one, and I am pretty sure can be safely deleted.

**Comment by enj** [comment]
@deads2k what are your thoughts on this?

**Comment by enj** [comment]
This will be used by the access restriction deny authorizer.

**Pull request closed without merging**